addons/web/controllers/main.py

```diff
@@ -505,11 +506,24 @@ def web_login(self, redirect=None, **kw):
         if not odoo.tools.config['list_db']:
             values['disable_database_manager'] = True
 
+        # otherwise no real way to test debug mode in template as ?debug =>
+        # values['debug'] = '' but that's also the fallback value when of
+        # missing variables in qweb
+        if 'debug' in values:
+            values['debug'] = True
+
         response = request.render('web.login', values)
         response.headers['X-Frame-Options'] = 'DENY'
         return response
 
+    @http.route('/web/become', type='http', auth='user', sitemap=False)
+    def switch_to_admin(self):
+        uid = request.env.user.id
+        if request.env.user._is_system():
+            uid = request.session.uid = odoo.SUPERUSER_ID
+            security.compute_session_token(request.session, request.env)
```

Comment:
That doesn't work for me, as the old token is still in cache and in the session.
We need to clear the cache, recompute token and store it in the session. Not super pretty, but heh... :man_shrugging: 
```
 request.env['res.users']._invalidate_session_cache()
 request.session.session_token = security.compute_session_token(request.session, request.env)
```

Reply:
Er… yeah forgot to reassign to session_token, bit of a brainfart here.